lib/sky-pages-route-generator.js

```diff
@@ -114,11 +129,30 @@ function generateDefinitions(routes) {
 function generateDeclarations(routes) {
   const p = indent(1);
   const declarations = routes
-    .map(r => `${p}{ path: '${r.routePath}', component: ${r.componentName} }`)
+    .map(r => {
+      let guard = r.guardPath && r.guardName ? `require('${r.guardPath}').${r.guardName}` : '';
```

Comment:
Was there not a way to build a list of classes to import like we do in the code where we detect components and add them to the declarations of the NgModule?  I'd like to use that same pattern if at all possible.  Same for the `generateProviders` method on line 148.